yt_dlp/extractor/rtp.py

```diff
@@ -65,16 +76,25 @@ def _real_extract(self, url):
             f_hls = f.get('hls')
             if f_hls is not None:
                 formats.extend(self._extract_m3u8_formats(
-                    f_hls, video_id, 'mp4', 'm3u8_native', m3u8_id='hls'))
+                    f_hls, video_id, 'mp4', 'm3u8_native', m3u8_id='hls',
+                    headers=self.__HEADERS))
 
             f_dash = f.get('dash')
             if f_dash is not None:
-                formats.extend(self._extract_mpd_formats(f_dash, video_id, mpd_id='dash'))
+                formats.extend(self._extract_mpd_formats(
+                    f_dash, video_id, mpd_id='dash',
+                    headers=self.__HEADERS))
+
+            for fmt in formats:
+                fmt.update({
+                    'http_headers': self.__HEADERS,
+                })
         else:
             formats.append({
                 'format_id': 'f',
                 'url': f,
                 'vcodec': 'none' if config.get('mediaType') == 'audio' else None,
+                'http_headers': self.__HEADERS,
```

Comment:
Since it applies to all formats, the headers can just be returned directly in the infodict

Reply:
?

Reply:
In `L108`
```py
        return {
            'id': video_id,
            'formats': formats,
            ...,
            'http_headers': self.__HEADERS
        }
```
This will get rid of `L85-89,95`